agenta-backend/agenta_backend/services/evaluation_service.py

```diff
@@ -798,3 +799,23 @@ def remove_duplicates(csvdata):
             unique_entries.append(entry)
 
     return unique_entries
+
+
+def fetch_evaluations_by_resource(resource_type: str, resource_ids: List[str]):
+    ids = list(map(lambda x: ObjectId(x), resource_ids))
+    if resource_type == "variant":
+        return EvaluationDB.find(In(EvaluationDB.variant, ids)).to_list()
+    elif resource_type == "testset":
+        return EvaluationDB.find(In(EvaluationDB.testset.id, ids)).to_list()
+    elif resource_type == "evaluator_config":
+        return EvaluationDB.find(
+            In(
+                EvaluationDB.evaluators_configs,
+                ids,
+            )
+        ).to_list()
```

Comment:
Could we `await` the queries we're making and make use of the `async` keyword in the function? I love your current approach, however, I'm concerned that it may not be easily understood by everyone.

Reply:
sure